packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart

```diff
@@ -107,6 +109,39 @@ class PurchaseParam {
 
   /// The 'sandboxTesting' is only available on iOS, set it to `true` for testing in AppStore's sandbox environment. The default value is `false`.
   final bool sandboxTesting;
+
+  /// The 'changeSubscriptionParam' is only available on Android, for upgrading or
+  /// downgrading an existing subscription. This does not require on iOS since
+  /// Apple provides a way to group related subscriptions together in iTunesConnect.
+  /// So when a subscription upgrade or downgrade is requested, Apple finds the old
+  /// subscription details from the group and handle it automatically.
+  final ChangeSubscriptionParam changeSubscriptionParam;
+}
+
+/// The parameter object for upgrading or downgrading an existing subscription so that
+/// the user can only purchase one subscription in a group at a time. This object
+/// is only applicable for Android.
+///
+/// This does not require on iOS since iTunesConnect provides a subscription grouping mechanism.
+/// Each subscription you offer must be assigned to a subscription group.
+/// So the developers can group related subscriptions together to prevent users from
+/// accidentally purchasing multiple subscriptions.
+///
+/// Please refer to the 'Creating a Subscription Group' sections of [Apple's subscription guide](https://developer.apple.com/app-store/subscriptions/)
+class ChangeSubscriptionParam {
+  /// Creates a new change subscription param object with given data
+  ChangeSubscriptionParam(
+      {@required this.oldPurchaseDetails, this.prorationMode});
+
+  /// The purchase object of the existing subscription that the user needs to
+  /// upgrade/downgrade from.
+  final PurchaseDetails oldPurchaseDetails;
```

Comment:
Can we not reduce the scope and accept just the `oldSku` / `oldProductID` instead, since that's the only parameter we need?

Reply:
+1

Reply:
I intentionally did this as mentioned in the description. 
While changing the subscription, this uses the old 'PurchaseDetails' object, rather than using 'ProductDetails' or just the 'oldSku' id itself, because, In the latest versions of the Play Billing Library, setOldSku() is deprecated (https://developer.android.com/reference/com/android/billingclient/api/BillingFlowParams.Builder#setoldsku)
and the new method uses the old purchase token. So it is better to use the 'PurchaseDetails' object here since it contains the 'VerificationData' to prepare for the next releases.

Reply:
didn't see your description about it! Now it makes more sense! And yeah I like it as is to prevent breaking change in the future.